src/elisp/treemacs-extensions.el

```diff
@@ -401,7 +402,7 @@ additional keys."
          (treemacs--button-close
           :button btn
           :new-state ',closed-state-name
-          :new-icon ,(if icon-closed closed-icon-name icon-closed-form)
+          :new-icon ,(unless (equal top-level-marker (quote 'variadic)) (if icon-closed closed-icon-name icon-closed-form))
```

Comment:
Don't start cleaning this up. I have plans to take care of that constant variadic switching (basically another define-node-type method that will delegate to this one).

Reply:
Ok, I added that `unless` because the icon switching function tries to update the current button, and fails with the hidden node.